src/Raven.Client/Documents/Session/ClusterTransactionSession.cs

```diff
@@ -391,11 +407,19 @@ internal ICommandData GetCommand(DocumentConventions conventions, JsonOperationC
                         var entity = CompareExchangeValueBlittableJsonConverter.ConvertToBlittable(_value.Value, conventions, context, jsonSerializer);
                         var entityJson = entity as BlittableJsonReaderObject;
                         BlittableJsonReaderObject metadata = null;
+                        _originalValue?.Value.TryGet(Constants.Documents.Metadata.Key, out metadata);
                         var metadataHasChanged = false;
                         if (_value.HasMetadata && _value.Metadata.Count != 0)
                         {
-                            metadata = PrepareMetadataForPut(_key, _value.Metadata, conventions, context);
-                            metadataHasChanged = InMemoryDocumentSessionOperations.UpdateMetadataModifications(_value?.Metadata, metadata);
+                            if (metadata == null)
+                            {
+                                metadataHasChanged = true;
+                                metadata = PrepareMetadataForPut(_key, _value.Metadata, conventions, context);
```

Comment:
`PrepareMetadataForPut` should be called for both cases, no?

Reply:
No, in case it already exist you only need to add the modifications to it